src/middlewared/middlewared/plugins/disk_/sync.py

```diff
@@ -98,6 +98,13 @@ class DiskService(Service, ServiceChangeMixin):
                         'iscsi.extent.query', [['type', '=', 'DISK'], ['path', '=', disk['disk_identifier']]]
                     ):
                         await self.middleware.call('iscsi.extent.delete', extent['id'])
+                    if disk['disk_kmip_uid']:
+                        try:
+                            await self.middleware.call('kmip.delete_kmip_secret_data', disk['disk_kmip_uid'])
```

Comment:
Does this contact KMIP server? Do we really need to wait for it to happen?

Reply:
It does and yes we don't. Will make required changes